parlai/scripts/train_model.py

```diff
@@ -227,21 +226,66 @@ def load_eval_world(agent, opt, datatype):
     if opt.get('eval_batchsize'):
         # override eval time batchsize
         opt['batchsize'] = opt['eval_batchsize']
-    if opt.get('validation_share_agent', False):
-        valid_agent = create_agent_from_shared(agent.share())
-    else:
-        valid_agent = agent
 
-    valid_world = create_task(opt, valid_agent)
-    return valid_world
+    tasks = opt['task'].split(',')
+    worlds = []
+    for task in tasks:
+        task_opt = opt.copy()  # copy opt since we edit the task
+        task_opt['task'] = task
+        if opt.get('validation_share_agent', False):
+            valid_agent = create_agent_from_shared(agent.share())
```

Comment:
this is actually a pretttty subtle change; i'm not sure it matters.

Previously (with multiworld running things). We're strictly creating more agents here than previously, and makes me wonder about our discussion in yesterday's meeting.

Reply:
i didn't change anything here

Reply:
it's within a loop; before it wasn't.